### PR TITLE
Bug 1424411 - document standardizing on submission_date

### PR DESCRIPTION
**Pull request by relud**
requesting @harterrt review as per contribution guidelines

@frank does this look like what you wanted for documentation in [bug 1424411](https://bugzilla.mozilla.org/show_bug.cgi?id=1424411#c3)?

### CODE REVIEW

File: concepts/submission_date.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+## Table of Contents
+
+<!-- toc -->
+
+# What is Submission Date
+
+`submission_date` is the server time at which a ping is received from the client. We use it to
+partition many of our data sets.
+
+# Submission Delay
```
Comment (fbertsch) [inline concepts/submission_date.md]: Make a note of what pings this is useful for. Are all custom pings using pingsender? Additionally, mobile products do not have pingsender, so they suffer delay, given in [0].

[0] https://sql.telemetry.mozilla.org/queries/49867#134105

File: concepts/submission_date.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Submission Date
+
+This document will help you understand `submission_date`, why are standardizing on it, and how it
+differs from `activity_date`.
+
+## Table of Contents
+
+<!-- toc -->
+
+# What is Submission Date
```
Comment (fbertsch) [inline concepts/submission_date.md]: Please describe here what `activity_date` is as well, which is based on `subsession_start_date`.

File: concepts/submission_date.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+a graph show 95% submission delay to generally sit between 100 and 150 hours before pingsender.
+[Bug 1336360](https://bugzilla.mozilla.org/show_bug.cgi?id=1336360) introduced pingsender and was
+merged in Firefox 55, which hit release on August 8, 2017. This means that using `submission_date`
+smears data between today and yesterday (mostly).
+
+# Why Submission Date
+
+TL;DR
+
+* not subject to client clock skew
```
Comment (fbertsch) [inline concepts/submission_date.md]: Nice tl;dr

File: concepts/submission_date.md (hunk 2 above)
Comment (relud) [reply concepts/submission_date.md]: fixed

File: concepts/submission_date.md (hunk 1 above)
Comment (relud) [reply concepts/submission_date.md]: fixed

File: concepts/submission_date.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Submission Date
+
+This document will help you understand `submission_date`, why are standardizing on it, and how it
+differs from `activity_date`.
+
+## Table of Contents
```
[the remark this answers is not in the document]
Comment (relud) [reply concepts/submission_date.md]: fixed

File: concepts/submission_date.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+## What are Submission Date and Activity Date
+
+`submission_date` is the server time at which a ping is received from the client. We use it to
+partition many of our data sets.
+
+`activity_date` is the client time at which a subsession started. It is acquired by removing time
+of day from `subsession_start_date`. It is sometimes normalized by throwing away values too far
+from `submission_date`, values not in the years 2000-2099, and values that are not dates.
+
+## Submission Delay
```
Comment (harterrt) [inline concepts/submission_date.md]: This is already covered in Analysis Gotchas, so if we combine these docs we can cut this section (or update the other blurb)

File: concepts/submission_date.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+* not subject to client clock skew
+* doesn't require normalization
+* good for backfill
+* good for daily processing
+* and usually good enough
+
+In [bug 1422892](https://bugzilla.mozilla.org/show_bug.cgi?id=1422892) we decided to standardize
+on `submission_date`. From [Comment 1](https://bugzilla.mozilla.org/show_bug.cgi?id=1422892#c1):
+
+> `activity_date` is generally preferable to `submission_date` because it's closer to what we
```
Comment (harterrt) [inline concepts/submission_date.md]: Not a strong opinion, but I'd suggest limiting this section to a recommendation and the TLDR, leaving the quote and reasoning in the bug.

File: concepts/submission_date.md (hunk 5 above)
Comment (relud) [reply concepts/submission_date.md]: I moved everything into Analysis Gotchas and updated the Delay blurb within

File: concepts/submission_date.md (hunk 6 above)
Comment (relud) [reply concepts/submission_date.md]: dropped the bug quotes

**Comment by harterrt** [comment]
@relud are we good to merge this PR?

**Comment by relud** [comment]
I was hoping for one more review from @fbertsch, but he's out this week, so sure, let's merge.

File: concepts/analysis_gotchas.md [hunk 7]
```diff
@@ -66,6 +66,28 @@ Build ids look like dates but aren't. If you take the first eight characters you
...
 
-A rule of thumb (pending [bug 1336360](https://bugzilla.mozilla.org/show_bug.cgi?id=1336360)) is data from two days ago is usually fairly representative.
+A rule of thumb is data from two days ago is usually fairly representative.
 
-If you'd like to read more about this subject and look at pretty graphs, there are a series of blogposts [here](https://chuttenblog.wordpress.com/2017/02/09/data-science-is-hard-client-delays-for-crash-pings/).
+If you'd like to read more about this subject and look at pretty graphs, there are a series of blogposts [here](https://chuttenblog.wordpress.com/2017/02/09/data-science-is-hard-client-delays-for-crash-pings/), [here](https://chuttenblog.wordpress.com/2017/07/12/latency-improvements-or-yet-another-satisfying-graph/) and [here](https://chuttenblog.wordpress.com/2017/09/12/two-days-or-how-long-until-the-data-is-in/).
+
+#### Pinsender
```
Comment (chutten) [inline concepts/analysis_gotchas.md]: typo: pingsender

**Pull request merged**